nixos/modules/system/boot/systemd.nix

```diff
@@ -606,6 +615,16 @@ in
     systemd.targets.network-online.wantedBy = [ "multi-user.target" ];
     systemd.services.systemd-importd.environment = proxy_env;
     systemd.services.systemd-pstore.wantedBy = [ "sysinit.target" ]; # see #81138
+    systemd.services.systemd-networkd-wait-online.serviceConfig.ExecStart = mkIf
+      (0 != length config.systemd.network-wait-online.ignore) [
```

Comment:
(nit) `config.systemd.network-wait-online.ignore != []` may be a little bit more lightweight to evaluate.